app/uk/gov/hmrc/softdrinksindustrylevy/services/MongoBufferService.scala

```diff
@@ -16,69 +16,90 @@
+        IndexModel(Indexes.ascending("status")),
+        IndexModel(Indexes.ascending("utr"))
+      )
+    ) {
+  // queries and updates can now be implemented with the available `collection: org.mongodb.scala.MongoCollection`
   def updateStatus(id: String, newStatus: String)(implicit ec: ExecutionContext): Future[Unit] =
-    collection.findAndUpdate(
-      BSONDocument("_id"  -> BSONString(id)),
-      BSONDocument("$set" -> BSONDocument("status" -> BSONString(newStatus)))
-    ) map { _ =>
+    collection
+      .findOneAndUpdate(
+        filter = Filters.equal("_id", id),
+        update = Updates.set("status", newStatus)
+      )
+      .toFuture() map { _ =>
       ()
     }
 
-  def findOverdue(createdBefore: Instant)(implicit ec: ExecutionContext): Future[Seq[SubscriptionWrapper]] = {
-    val bsonDt = BSONDateTime(createdBefore.toEpochMilli)
-    find("status" -> "PENDING", "timestamp" -> Json.obj("$lt" -> bsonDt))
-  }
+  def findOverdue(createdBefore: LocalDateTime)(implicit ec: ExecutionContext): Future[Seq[SubscriptionWrapper]] =
+    collection
+      .find(
+        Filters.and(
+          Filters.equal("status", "PENDING"),
+          Filters.lt("timestamp", createdBefore)
+        ))
+      .toFuture()
+
+  def insert(sub: SubscriptionWrapper)(implicit ec: ExecutionContext): Future[Unit] =
+    collection.insertOne(sub).toFuture() map (_ => ())
+
+  def find(utr: String)(implicit ec: ExecutionContext): Future[Seq[SubscriptionWrapper]] =
+    collection.find(Filters.equal("utr", utr)).toFuture()
+
+  def remove(utr: String)(implicit ec: ExecutionContext): Future[DeleteResult] =
+    collection.deleteOne(Filters.equal("utr", utr)).toFuture()
+
+  def findById(id: String)(implicit ec: ExecutionContext): Future[SubscriptionWrapper] =
+    collection.find(Filters.equal("_id", id)).toFuture() map (_.head)
+
+  def removeById(id: String)(implicit ec: ExecutionContext): Future[DeleteResult] =
+    collection.deleteOne(Filters.equal("_id", id)).toFuture()
 
-  override def indexes: Seq[Index] = Seq(
-    Index(
-      key = Seq("timestamp" -> IndexType.Ascending),
-      name = Some("ttl"),
-      options = BSONDocument("expireAfterSeconds" -> (30 days).toSeconds)
-    )
-  )
 }
 
 case class SubscriptionWrapper(
   _id: String,
   subscription: Subscription,
   formBundleNumber: String,
-  timestamp: Instant = Instant.now,
+  timestamp: LocalDateTime = LocalDateTime.now,
   status: String = "PENDING")
 
 object SubscriptionWrapper {
   implicit val subFormat: Format[Subscription] = Format(subReads, subWrites)
 
-  implicit val instantFormat: Format[Instant] = new Format[Instant] {
-    override def writes(o: Instant): JsValue =
-      Json.toJson(BSONDateTime(o.toEpochMilli))
-
-    override def reads(json: JsValue): JsResult[Instant] =
-      json.validate[BSONDateTime] map { dt =>
-        Instant.ofEpochMilli(dt.value)
-      }
+  implicit val localDateTimeFormat: Format[LocalDateTime] = new Format[LocalDateTime] {
```

Comment:
I think for java LocalDateTime we don't need formats, it's there by default . Can you please double check